controllers/job_mgr/job_mgr_controller.go

```diff
@@ -0,0 +1,177 @@
+	return job.Spec.Suspend
+}
+
+func (job *LMEvalJob) Suspend() {
+	job.Spec.Suspend = true
+}
+
+// RunWithPodSetsInfo will inject the node affinity and podSet counts extracting from workload to job and unsuspend it.
+func (job *LMEvalJob) RunWithPodSetsInfo(podSetsInfo []podset.PodSetInfo) error {
+	job.Spec.Pod.Affinity = convertToAffinity(podSetsInfo)
+	job.Spec.Suspend = false
+	return nil
+}
+
+// RestorePodSetsInfo will restore the original node affinity and podSet counts of the job.
+// Returns whether any change was done.
+func (job *LMEvalJob) RestorePodSetsInfo(podSetsInfo []podset.PodSetInfo) bool {
+	job.Spec.Pod.Affinity = convertToAffinity(podSetsInfo)
+	job.Spec.Suspend = false
+	return true
+}
+
+// Finished means whether the job is completed/failed or not,
+// condition represents the workload finished condition.
+
+func (job *LMEvalJob) Finished() (condition metav1.Condition, finished bool) {
+	// ToDos: What should be in the condition and how will it be used?
+	condition = metav1.Condition{
+		Type:    "Finished",
+		Status:  "False",
+		Reason:  string(job.Status.Reason),
+		Message: job.Status.Message,
+	}
+	finished = false
+
+	if job.Status.State == workloadv1alpha1.CompleteJobState {
+		condition.Status = "True"
+		finished = true
+		return
+	}
+	return
+}
+
+// PodSets will build workload podSets corresponding to the job.
+func (job *LMEvalJob) PodSets() []kueue.PodSet {
+	log := log.FromContext(context.TODO())
+	pod := lmes.CreatePod(lmes.Options, &job.LMEvalJob, log)
+	podSet := kueue.PodSet{
+		Name:     job.GetPodName(),
+		Count:    1,
+		Template: corev1.PodTemplateSpec{Spec: pod.Spec},
+	}
+	podSets := []kueue.PodSet{}
+	podSets = append(podSets, podSet)
+	return podSets
+}
+
+// IsActive returns true if there are any running pods.
+func (job *LMEvalJob) IsActive() bool {
+	if job.Status.State == workloadv1alpha1.RunningJobState {
+		return true
+	} else {
+		return false
+	}
+}
+
+// PodsReady instructs whether job derived pods are all ready now.
+func (job *LMEvalJob) PodsReady() bool {
+	if job.Status.State == workloadv1alpha1.ScheduledJobState {
+		return true
+	} else {
+		return false
+	}
+}
+
+// GVK returns GVK (Group Version Kind) for the job.
+func (job *LMEvalJob) GVK() schema.GroupVersionKind {
+	return workloadv1alpha1.GroupVersion.WithKind("LMEvalJob")
+}
+
+// Convert NodeSelector in the PodSetInfo to Pod.Spec.Affinity
+func convertToAffinity(psi []podset.PodSetInfo) *corev1.Affinity {
+	if len(psi) > 0 {
+		nsl := psi[0].NodeSelector // Note there is only 1 element in podset array see PodSets method above.
+		nsra := []corev1.NodeSelectorRequirement{}
+		for k, v := range nsl {
+			nsr := corev1.NodeSelectorRequirement{
+				Key:      k,
+				Operator: "In",
+				Values:   []string{v},
+			}
+			nsra = append(nsra, nsr)
+		}
+		nsta := []corev1.NodeSelectorTerm{}
+		nsta = append(nsta, corev1.NodeSelectorTerm{MatchExpressions: nsra})
+		affinity := corev1.Affinity{
```

Comment:
nit: you can return here
```suggestion
		return &corev1.Affinity{
```
and remove the return at #174